tests/playwright/src/model/pages/main-page.ts

```diff
@@ -94,4 +94,15 @@ export abstract class MainPage extends BasePage {
     }
     return undefined;
   }
+
+  async countRowsFromTable(): Promise<number> {
+    try {
+      await this.page.waitForSelector('[role="row"]', { state: 'visible', timeout: 1000 });
+      const table = await this.getTable();
+      const rows = await table.getByRole('row').all();
+      return rows.length - 1; //counts the header of the table as a row, that's why one is substracted
+    } catch (error) {
+      return 0; //there are 0 visible rows
```

Comment:
I don't really understand this implementation, why do you need to wait for Selector? Especially since this requires a ugly solution in the try catch. Doesn't the all() method return 0 if there is nothing there? I assume in this case it should return 1 since there is the header, so why can't you just return the count of all()?

Reply:
I tried a similar implementation to getRowFromTableByName(), using only all(), but I kept getting the wrong number of rows because the page takes a second or so to load the table that contains the volumes (specially if the volumes are created manually, not sure why), so if you navigate to the volumes page or prune the volumes and don't give it a little bit of time, you get inconsistent results. There has to be a better way to do it but I'm not sure how.
[not waiting for row.webm](https://github.com/user-attachments/assets/0a6225db-2c66-40a1-9a69-04c3719fcc15)
[waiting for row.webm](https://github.com/user-attachments/assets/0b52808b-588b-40c6-8014-84b676e44b04)

Reply:
Create a boolen method that returns the value of `await this.page.getByRole('row').first().isVisible()` then use that method as a callback in the waitUntil() method while setting the `sendError` param of waitUntil to `false`, this way you can wait for the header row to be visible without throwing an exception and needing the try catch to handle it.